services/algolia-sync/src/components/algolia/site-sync.js

```diff
@@ -21,6 +21,7 @@ const query = gql`
           type
           created
           published
+          unpublishedDate
```

Comment:
This should use the `unpublished` field - as it will return a integer timestamp (in milliseconds) if set, otherwise `null`

The `unpublishedDate` returns a string, which would cause your data types to be mis-matched